From 3fe67deb9fcf0ae3c2ff31a9eccb6a0a9af33b9c Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 14 Jan 2022 09:47:03 -0800
Subject: [PATCH] test: Disable optimization with clang for aligned_alloc.c

Clang crashes when using glibc 2.35, it works ok with older glibc or
musl, so its very specific problem. Its reported here

https://github.com/llvm/llvm-project/issues/52765

Until it is fixed, workaround the build failure

Upstream-Status: Inappropriate [Workaround]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 test/integration/aligned_alloc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/integration/aligned_alloc.c b/test/integration/aligned_alloc.c
index 4375b172..6513bdd9 100644
--- a/test/integration/aligned_alloc.c
+++ b/test/integration/aligned_alloc.c
@@ -1,5 +1,7 @@
 #include "test/jemalloc_test.h"
 
+#pragma clang optimize off
+
 #define MAXALIGN (((size_t)1) << 23)
 
 /*
-- 
2.34.1

